ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13720,15 +13720,7 @@ ASTNode analyzeCreateTable(
         }
         break;
       case HiveParser.TOK_TABLEPARTCOLSBYSPEC:
-        List<TransformSpec> partitionTransformSpec =
-            PartitionTransform.getPartitionTransformSpec(child);
-
-        if (!SessionStateUtil.addResource(conf, hive_metastoreConstants.PARTITION_TRANSFORM_SPEC,
-            partitionTransformSpec)) {
-          throw new SemanticException("Query state attached to Session state must be not null. " +
-              "Partition transform metadata cannot be saved.");
-        }
-
```

Comment:
Since the `if` block appears in at least three places consider refactoring it independently. It could be done as a more generic method in `SessionStateUtil`.

```java
public static void addResourceOrThrow(Configuration conf, String key, Object resource) {
    Optional<QueryState> queryState = getQueryState(conf);
    if (queryState.isPresent()) {
      queryState.get().addResource(key, resource);
    } else {
      throw new IllegalStateException("Query state is missing; failed to add resource for " + key);
    }
  }
```
Then its up to you if you want to introduce `parseAndStorePartitionTransformSpec` or not.